app/src/main/java/com/github/sdp/mediato/CreateProfileFragment.java

```diff
@@ -0,0 +1,136 @@
+package com.github.sdp.mediato;
+
+import android.annotation.SuppressLint;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+
+import android.text.Editable;
+import android.text.InputFilter;
+import android.view.KeyEvent;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+
+import com.github.dhaval2404.imagepicker.ImagePicker;
+import com.github.javafaker.Faker;
+import com.google.android.material.button.MaterialButton;
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
+import com.google.android.material.textfield.TextInputEditText;
+import com.google.android.material.textfield.TextInputLayout;
+
+/**
+ * create an instance of this fragment.
+ */
+public class CreateProfileFragment extends Fragment {
+
+    private ImageView profileImage;
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_create_profile, container, false);
+
+        final TextInputLayout usernameTextInput = view.findViewById(R.id.username_text_input);
+        final TextInputEditText usernameEditText = view.findViewById(R.id.username_edit_text);
+        final MaterialButton createButton = view.findViewById(R.id.create_profile_button);
+        final FloatingActionButton profileImageButton = view.findViewById(R.id.profile_image_add_button);
+        profileImage = view.findViewById(R.id.profile_image);
+
+        profileImageButton.setOnClickListener(photoPicker());
+        usernameTextInput.setEndIconOnClickListener(generateUsername(usernameTextInput, usernameEditText));
+        usernameEditText.setOnKeyListener(removeErrorIfUsernameValid(usernameTextInput, usernameEditText));
+
+        //Set an error if the password is less than 8 characters.
+        createButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                int error = 0;
+                String errorMsg;
+                if (0 != (error = isPasswordValid(usernameEditText.getText()))) {
+                    errorMsg = (error == 1) ? getString(R.string.mt_username_error_1)
+                            : (error == 2) ? getString(R.string.mt_username_error_2)
+                            : getString(R.string.mt_username_error_3);
+
+                    usernameTextInput.setError(errorMsg);
+                } else {
+                    usernameTextInput.setError(null);
+                    //((NavigationHost) getActivity()).navigateTo(new ProductGridFragment(), false);
```

Comment:
Is there a particular reason why this bit of code is commented ? Can it be removed ?

Reply:
Yeah because the next page doesn't exist